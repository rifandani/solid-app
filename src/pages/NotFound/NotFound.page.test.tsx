import { Route } from '@solidjs/router';
import { waitFor } from '@solidjs/testing-library';
import { renderWithRouter } from '../../utils/test.util';
import NotFoundPage from './NotFound.page';

describe('NotFoundPage', () => {
  it('should render correctly', async () => {
    // ARRANGE
    const { container, getByText } = renderWithRouter(() => (
      <Route path="/" component={NotFoundPage} /> // the path doesn't matter here
    ));

    // ASSERT
    await waitFor(() => {
      expect(getByText(/404: Not Found/)).toBeInTheDocument();
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
