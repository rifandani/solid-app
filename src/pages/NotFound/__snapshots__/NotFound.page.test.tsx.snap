// Vitest Snapshot v1

exports[`NotFoundPage > should render correctly 1`] = `
<section
  class="text-gray-700 p-8"
>
  <h1
    class="text-2xl font-bold"
  >
    404: Not Found
  </h1>
  <p
    class="mt-4"
  >
    It's gone 😞
  </p>
</section>
`;
